python/plugins/processing/algs/gdal/extractprojection.py

```diff
@@ -47,8 +47,8 @@ def __init__(self):
         super().__init__()
 
     def initAlgorithm(self, config=None):
-        self.addParameter(ParameterRaster(self.INPUT, self.tr('Input file')))
-        self.addParameter(ParameterBoolean(self.PRJ_FILE,
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT, self.tr('Input file')))
+        self.addParameter(QgsProcessingParameterBoolean(self.PRJ_FILE,
```

Comment:
It might be nice to also add a file output here:

   self.addOutput(QgsProcessingOutputFile(self.WORLD_FILE, self.tr('World file')))

and return the path to the newly created world file within the algorithm results. (Maybe also the optional prj file too?) This would make this algorithm more useful within a model context, as these filenames could be used with other later algorithms, such as copying to a different location, etc.

Reply:
I have tried.